istioctl/pkg/auth/validator.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"fmt"
+	"strings"
+
+	rbacproto "istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+type Validator struct {
+	PolicyFiles          []string
+	RoleKeyToServiceRole map[string]model.Config
+	serviceRoleBindings  []model.Config
+	Report               strings.Builder
+}
+
+const (
+	RoleNotFound = "serviceRoleNotFound: %q used by ServiceRoleBinding %q at namespace %q\n"
```

Comment:
I think it's not good practice to use format string like this, you can wrap this in a function so the argument list tells which parameter is for what.

Reply:
Sg. Will change